### PR TITLE
products and filters added

### PR DESCRIPTION
**Pull request by shrey27**
Added Features

- Landing page
- products page
- filters
- search bar

### CODE REVIEW

File: src/frontend/common/deals/Deal.js [hunk 1]
```diff
@@ -0,0 +1,151 @@
+import './deals.css';
+// import { useCartCtx, useCartAPICtx } from '../../context/cartContext';
+// import { useWishlistCtx } from '../../context/wishlistContext';
```
Comment (Rohanmond) [inline src/frontend/common/deals/Deal.js:3]: delete these comments

File: src/frontend/common/deals/Deal.js [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+    fastdelivery,
+    count,
+    nostock,
+    wishlist,
+    close,
+    noButton
+  } = props;
+
+  // const { addToCart } = useCartCtx();
+  // const { addToWishlist, deleteFromWishlist, addedPID } = useWishlistCtx();
```
Comment (Rohanmond) [inline src/frontend/common/deals/Deal.js:27]: delete these comments also

File: src/frontend/common/deals/Deal.js [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+    source,
+    title,
+    price,
+    mrp,
+    discount,
+    rating,
+    fastdelivery,
+    count,
+    nostock,
+    wishlist,
```
Comment (Rohanmond) [inline src/frontend/common/deals/Deal.js:21]: If possible do these things using context, try not to pass these may props

File: src/frontend/common/deals/index.js [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+import Deal from './Deal';
+
+export default function Deals({ items, name, wishlist, close, noButton }) {
+  return (
+    <section className='deals'>
+      <h1 className='primary lg xs-s cen'>{name}</h1>
+      <div className='deal--cards xs-s'>
+        {items.map((elem, index) => {
+          return (
+            <Deal key={index * 2} {...elem} wishlist={wishlist} close={close} noButton={noButton}/>
```
Comment (Rohanmond) [inline src/frontend/common/deals/index.js:11]: key can be elem id

**Review by Rohanmond — APPROVED** [review]

File: src/frontend/common/constants.js [hunk 5]
```diff
@@ -0,0 +1,586 @@
...
+    source:
+      'https://s1.thcdn.com/productimg/300/300/12708205-7894888116404280.jpg',
+    title: 'Protein Starter Pack',
+    price: 5000,
+    discount: 30,
+    rating: 4.2
+  }
+];
+
+// export const items = [
```
Comment (rohit8686) [inline src/frontend/common/constants.js:68]: Folder structure looks good but try to remove commented codes

**Review by rohit8686 — APPROVED** [review]

**Pull request merged**